script.js

```diff
@@ -1,12 +1,273 @@
+
+       
+ // Info regarding TVMaze.com
+ // EG <a href="www.tvmaze.com">The movies displayed on this site have been sourced from TVMaze.com</a>
+   
+  tvmInfoDiv = document.createElement('div');
+  let paragraph = document.createElement('span');
+  paragraph.innerText = "Please note: The movies displayed on this site have been sourced from ";
+  tvmInfoDiv.appendChild(paragraph); 
+  let theLink = document.createElement("a");
+  theLink.href = "https://www.tvmaze.com/";        
+  theLink.textContent = "TVMaze.com";
+  theLink.setAttribute('target', '_blank'); // Open in another tab
+  
+  // Recommended setting: https://www.freecodecamp.org/news/how-to-use-html-to-open-link-in-new-tab/
+  theLink.setAttribute('rel', 'noreferrer noopener');
+  
+  tvmInfoDiv.appendChild(theLink);
+  tvMazeInfo.appendChild(tvmInfoDiv);
 }
 
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
   rootElem.textContent = `Got ${episodeList.length} episode(s)`;
 }
 
-window.onload = setup;
+function showAll(setup_options) {
+       for (let i = 0; i < allEpisodes.length; i++)
+       {
+              createAllEpisodes(i);
+              // at the same time set up the dropdown option menu
+              if (setup_options) { 
+                          let option = document.createElement('option');
+                          option.value = fetchEpisodeSeason_Suffix(allEpisodes[i]); // EG 'S01E01'
+                          option.text = fetchEpisodeSeason_Full(allEpisodes[i]); // EG Winter is Coming - S01E01
+                          selectMenu.appendChild(option);
+                                }           
+       }
+       displayMessage[0].innerText = `Displaying ${allEpisodesTotal}/${allEpisodesTotal} episodes.`;
+}
+
+function createAllEpisodes(index) {
+    const episodeDiv = document.createElement('div'); // main 'div' to append to
+    const source = allEpisodes[index]; // Entire Episode Entry
+    episodeDiv.setAttribute("id",fetchEpisodeSeason_Suffix(source)); // EG S01E01
+
+    let episodeInfo = fetchEpisodeSeason_Full(source); // EG Winter is Coming - S01E01
+    const text = document.createElement('h1');
+    text.classList.add('episode-title');
+    text.innerText = episodeInfo;
+    episodeDiv.appendChild(text); // add the title
+
+    const imageDiv = document.createElement('div'); // div for an image
+    imageDiv.classList.add('image');
+    episodeDiv.appendChild(imageDiv);
+
+    const img = document.createElement('img');
+    img.src = source.image.medium;
+    img.alt = episodeInfo;
+    episodeDiv.appendChild(img); // add the image
+
+    // Remove all <p></p> then add the text using <span>
+    const summary = handle_paragraph(source.summary);
+    episodeDiv.appendChild(summary); // add the summary
+
+    // add to main content
+    mainDisplayDiv.appendChild(episodeDiv); 
+
+    if ((index+1) % 3 === 0) // MAKE THE 4TH COLUMN A GAP BY CREATING AN EMPTY DIV
```

Comment:
You already have a class .gridDisplay, which has padding-left: 3em
As an option, you could do the same for padding-right instead of an empty <did>

Reply:
Thank you for the other solution that you showed me